packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
 import { Controller } from 'ember-runtime';
-import { run } from 'ember-metal';
-
-import { Application } from 'ember-application';
-import { Router } from 'ember-routing';
-import { compile } from 'ember-template-compiler';
-import {
-  Component,
-  setTemplates,
-  setTemplate
-} from 'ember-glimmer';
-import { jQuery } from 'ember-views';
-
-let App, appInstance;
-
-function prepare() {
-  setTemplate('components/expand-it', compile('<p>hello {{yield}}</p>'));
-  setTemplate('application', compile('Hello world {{#expand-it}}world{{/expand-it}}'));
-}
-
-function cleanup() {
-  run(() => {
-    try {
-      if (App) {
-        App.destroy();
-      }
-      App = appInstance = null;
-    } finally {
-      setTemplates({});
-    }
-  });
-}
-
-QUnit.module('Application Lifecycle - Component Registration', {
-  setup: prepare,
-  teardown: cleanup
-});
-
-function boot(callback, startURL = '/') {
-  run(() => {
-    App = Application.create({
-      name: 'App',
-      rootElement: '#qunit-fixture'
-    });
+import { Component } from 'ember-glimmer';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 
-    App.deferReadiness();
+moduleFor('Application Lifecycle - Component Registration', class extends ApplicationTestCase {
 
-    App.Router = Router.extend({
-      location: 'none'
+  ['@test The helper becomes the body of the component'](assert) {
+    this.addTemplate('application', 'Hello world {{#expand-it}}world{{/expand-it}}');
+    this.addComponent('expand-it', {
+      template: '<p>hello {{yield}}</p>'
     });
 
-    appInstance = App.__deprecatedInstance__;
-
-    if (callback) { callback(); }
-  });
-
-  let router = appInstance.lookup('router:main');
-
-  run(App, 'advanceReadiness');
-  run(() => router.handleURL(startURL));
-}
-
-QUnit.test('The helper becomes the body of the component', function() {
-  boot();
-  equal(jQuery('div.ember-view > div.ember-view', '#qunit-fixture').text(), 'hello world', 'The component is composed correctly');
-});
-
-QUnit.test('If a component is registered, it is used', function() {
-  boot(() => {
-    appInstance.register('component:expand-it', Component.extend({
```

Comment:
You'll need to use an autoboot application test case for this. The test here is that registering a component onto the application instance inside the runloop where that application is created still loads that component.

```js
moduleFor('Application Lifecycle - Component Registration w/ autoboot', class extends AutobootApplicationTestCase {

  [`@test If a component is registered, it is used'](assert) {
    this.runTask(() => {
      this.createApplication();
      this.addTemplate('application', `Hello world {{#expand-it}}world{{/expand-it}}`);
      this.applicationInstance.register('component:expand-it', Component.extend({
        classNames: 'testing123'
      });
    });
    /* ... assertion ... */
  }

});
```

In this test the application template is added to the resolver, but the component is registered on the application instance.